wordpress_org_api/tests/test_plugin_directory.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    env,
+    io::{self, Write},
+};
+
+use serde::Deserialize;
+use wordpress_org_api::plugin_directory::*;
+
+async fn query_plugins_slugs(url: &str) -> Result<Vec<String>, reqwest::Error> {
+    #[derive(Deserialize, Debug)]
+    struct Response {
+        plugins: Vec<Plugin>,
+    }
+    #[derive(Deserialize, Debug)]
+    struct Plugin {
+        slug: String,
+    }
+
+    reqwest::get(url)
+        .await?
+        .json::<Response>()
+        .await?
+        .plugins
+        .iter()
+        .map(|p| Ok(p.slug.clone()))
+        .collect()
+}
+
+async fn plugin_information(slug: &str) -> Result<PluginInformation, reqwest::Error> {
+    let url = format!(
+        "https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request[slug]={}&fields=icons",
+        slug
+    );
+    reqwest::get(&url)
+        .await?
+        .json::<PluginInformation>()
+        .await
+        .map_err(Into::into)
+}
+
+#[tokio::test]
+async fn test_parsing_full_plugin_directory() {
+    let page_size: u64;
+    let total_pages: u64;
+    if env::var("TEST_ALL_PLUGINS").is_ok() {
+        println!("Checking how many pages to fetch...");
+        page_size = 200;
+        let url = format!(
+            "https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[per_page]={}",
+            page_size
+        );
+        let response = reqwest::get(&url).await.unwrap();
+        let json = response.json::<serde_json::Value>().await.unwrap();
+        total_pages = json["info"]["pages"].as_u64().unwrap();
+    } else {
+        println!("Only a small amount of plugins will be fetched.");
+        page_size = 10;
+        total_pages = 2;
+    }
+
+    let mut query_plugins_failures = Vec::new();
+    let mut all_slugs: Vec<String> = Vec::new();
+    for page in 1..=total_pages {
```

Comment:
IMHO we shouldn't hit the endpoint at all if we're not running a full test.

I see you have a `json` file for testing, can we have one of those for this so that it runs quickly? We're just interested in testing the parsing code, so I'd love to factor out networking for the vast majority of cases if we can

Reply:
> IMHO we shouldn't hit the endpoint at all if we're not running a full test.

Considering this test is part of integration tests, I think it's okay for it to make a small amount of HTTP requests? There is still value in running this test during PR (and locally development), like running the entire test function to ensure there is not obvious issue in the test code itself.

> I see you have a json file for testing, can we have one of those for this so that it runs quickly?
I have added one in https://github.com/Automattic/wordpress-rs/pull/424/commits/022fa1f4b22fcac33b6a2af4721c6e1746b7f6c8 as part of unit tests.